src/integrations/admin/indexing-notification-integration.php

```diff
@@ -162,13 +162,13 @@ public static function get_conditionals() {
 	 * it to the notification center if this is the case.
 	 */
 	public function create_notification() {
-		$notification = $this->notification_center->get_notification_by_id( self::NOTIFICATION_ID );
-
-		if ( $notification || ! $this->should_show_notification() ) {
+		if ( ! $this->should_show_notification() ) {
 			return;
 		}
 
-		$this->notification_center->add_notification( $this->notification() );
+		$notification = $this->notification();
+		$this->notification_center->restore_notification( $notification );
```

Comment:
I'm getting a warning "Static method 'restore_notification' should not be called dynamically".

Reply:
And what happens when this code is executed when there is nothing to restore? Do we maybe need a conditional here?

Reply:
Regarding the first question: we made a notification helper that wraps the static method `restore_notification`.
Regarding the second question: there is nothing to worry about, because when there is nothing to restore, the `restore_notification` function simply returns `false`.